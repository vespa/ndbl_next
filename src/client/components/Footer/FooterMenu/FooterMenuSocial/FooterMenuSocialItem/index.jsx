
import React from 'react';
import PropTypes from 'prop-types';
import cssMain from '../../../../../../scss/main.scss';

const FooterMenuSocialSocialItem = ({ text, icon, href }) => {
  const classe = `icon__social--${icon}`;
  return (
    <li>
      <span className={`${cssMain[classe]} ${cssMain.icon__social} ${cssMain.icon}`}> {text} </span>
      <a href={href}> {text} </a>
    </li>
  );
};

FooterMenuSocialSocialItem.propTypes = {
  text: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
};

export default FooterMenuSocialSocialItem;
