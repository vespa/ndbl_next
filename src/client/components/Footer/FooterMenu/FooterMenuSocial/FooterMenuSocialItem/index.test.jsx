import React from 'react';
import { shallow } from 'enzyme';
import FooterMenuSocialItem from './index';

describe('[Component: FooterMenuSocialItem]', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<FooterMenuSocialItem text="test" href="/" />);
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
