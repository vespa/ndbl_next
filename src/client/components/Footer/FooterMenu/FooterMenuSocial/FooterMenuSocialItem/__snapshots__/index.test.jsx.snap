// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component: FooterMenuSocialItem] should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FooterMenuSocialSocialItem
    href="/"
    text="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="undefined undefined undefined"
        >
           
          test
           
        </span>,
        <a
          href="/"
        >
           
          test
           
        </a>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            "test",
            " ",
          ],
          "className": "undefined undefined undefined",
        },
        "ref": null,
        "rendered": Array [
          " ",
          "test",
          " ",
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            "test",
            " ",
          ],
          "href": "/",
        },
        "ref": null,
        "rendered": Array [
          " ",
          "test",
          " ",
        ],
        "type": "a",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="undefined undefined undefined"
          >
             
            test
             
          </span>,
          <a
            href="/"
          >
             
            test
             
          </a>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "test",
              " ",
            ],
            "className": "undefined undefined undefined",
          },
          "ref": null,
          "rendered": Array [
            " ",
            "test",
            " ",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "test",
              " ",
            ],
            "href": "/",
          },
          "ref": null,
          "rendered": Array [
            " ",
            "test",
            " ",
          ],
          "type": "a",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
